linkedlist.c

```diff
@@ -0,0 +1,61 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+struct node
+{
+int data;
+struct node *next;
+};
+
+struct node *head=NULL;
+
+
+void insertion_beginning()
+{
+struct node *newNode;
+newNode =NULL;
+newNode=(struct node*)malloc(sizeof(struct node));
+printf("enter data :");
+scanf("%d",&newNode->data);
+if(head==NULL)
+{
+head=newNode;
+}
+else
+{
+newNode->next=head;
+head=newNode;
+}
+printf("\nNode inserted\n");
+}
+
+void display()
+{
+int count=0;
+struct node *newNode;
+printf("Available data:\n");
+newNode=head;
+while(newNode!=NULL)
+{
+printf("%d\n",newNode->data);
+count++;
+newNode=newNode->next;
+}
+printf("\nTotal nodes:%d\n",count);
+}
+
+int main()
+{
+int choice;
+ void (*ptr [3])()={ insertion_beginning, display,exit};
+while(1)

```

Comment:
while loop will always be true, So what happens if I give non integer number as a choice?

Reply:
It's also work fine.